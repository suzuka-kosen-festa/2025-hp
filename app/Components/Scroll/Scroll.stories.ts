import type { Meta, StoryObj } from "@storybook/react";

import Scroll from "./Scroll";

type T = typeof Scroll;

const meta: Meta = {
  title: "Scroll",
  component: Scroll,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} satisfies Meta<T>;

type Story = StoryObj<T>;

export const Default: Story = {};

export default meta;
