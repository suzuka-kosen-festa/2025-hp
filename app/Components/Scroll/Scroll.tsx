import type { ReactNode } from "react";
import { Box, keyframes, styled } from "@mui/material";

const scrolldown = keyframes`
  0% {
    transform: translateY(0);
  }
  30% {
    transform: translateY(10px);
  }
`;

const Scrolltext = styled("span")(({ theme }) => ({
  position: "absolute",
  bottom: "10px",
  left: "-40px",
  color: theme.palette.text.primary,
  fontSize: "1.2rem",
  letterSpacing: "0.1em",
  textTransform: "uppercase",
  writingMode: "vertical-rl",
}));

const ScrollDown = styled("div")({
  "position": "absolute",
  "bottom": 0,
  "left": "50%",
  "animation": `${scrolldown} 1.5s infinite`,
  "&::before": {
    content: "\"\"",
    position: "absolute",
    bottom: "30px",
    left: "-3px",
    width: "6px",
    height: "20px",
    background: "black",
    borderRadius: "4px",
  },
  "&::after": {
    content: "\"\"",
    position: "absolute",
    bottom: "20px",
    left: "-6px",
    width: "12px",
    height: "12px",
    borderBottom: "2px solid black",
    borderRight: "2px solid black",
    transform: "rotate(45deg)",
  },
});

function Scroll(): ReactNode {
  return (
    <Box
      sx={{
        position: "absolute",
        bottom: "20px",
        right: "20px",
      }}
    >
      <Scrolltext>Scroll</Scrolltext>
      <ScrollDown />
    </Box>
  );
}

export default Scroll;
