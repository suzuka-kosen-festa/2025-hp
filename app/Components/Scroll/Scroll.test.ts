// @vitest-environment jsdom

import { composeStories } from "@storybook/react";
import { expect, it } from "vitest";

import * as stories from "./Scroll.stories";

const { Default } = composeStories(stories);

it("scroll snapshot", async () => {
  await Default.run();
  expect(document.body.firstChild).toMatchSnapshot();
});
