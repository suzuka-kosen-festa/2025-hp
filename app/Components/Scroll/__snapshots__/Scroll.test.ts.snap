// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`scroll snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1kue2vf"
  >
    <span
      class="css-1pnf43l"
    >
      Scroll
    </span>
    <div
      class="css-upika3"
    />
  </div>
</div>
`;
